xtable-core/src/main/java/org/apache/xtable/delta/DeltaStatsExtractor.java

```diff
@@ -229,6 +286,114 @@ public FileStats getColumnStatsForFile(AddFile addFile, List<InternalField> fiel
     }
   }
 
+  /**
+   * Reads column statistics directly from a Parquet file footer. This method is used as a fallback
+   * when Delta checkpoint statistics are NULL or unavailable.
+   *
+   * <p>This operation is expensive as it requires:
+   *
+   * <ul>
+   *   <li>Opening each Parquet file individually (I/O overhead)
+   *   <li>Reading the file footer metadata
+   *   <li>Parsing column chunk metadata for all columns
+   *   <li>Converting Parquet statistics to internal format
+   * </ul>
+   *
+   * <p>For cloud storage (S3, GCS, ADLS), this can add significant latency due to network overhead.
+   * The method performs several safety checks to prevent errors:
+   *
+   * <ul>
+   *   <li>Filters out statistics with NULL min/max ranges (prevents NullPointerException)
+   *   <li>Skips DECIMAL and complex types (prevents ClassCastException)
+   *   <li>Validates Binary-to-primitive type conversions
+   * </ul>
+   *
+   * <p>Record Count: The record count is read from Parquet row group metadata, which is always
+   * reliable regardless of column statistics availability.
+   *
+   * @param addFile the Delta AddFile action containing the file path
+   * @param snapshot the Delta snapshot providing table base path
+   * @param fields the schema fields for which to extract statistics
+   * @return FileStats with extracted statistics, or empty stats if reading fails
+   */
+  private FileStats readStatsFromParquetFooter(
```

Comment:
Let's also make sure there is a test case to cover this flow. It can be a basic one, the detailed coverage for translating from parquet stats to our internal stats representation will be covered in https://github.com/apache/incubator-xtable/issues/748